tools/server-spring/src/main/java/org/eclipse/rdf4j/http/server/repository/handler/DefaultQueryRequestHandler.java

```diff
@@ -75,6 +80,22 @@ public DefaultQueryRequestHandler(RepositoryResolver repositoryResolver) {
 		super(repositoryResolver);
 	}
 
+	@Override
+	protected Explanation explainQuery(final Query query, final Explanation.Level level) {
+		return query.explain(level);
+	}
+
+	@Override
+	protected ModelAndView getExplainQueryResponse(
+			final HttpServletRequest request, final HttpServletResponse response,
+			final Explanation explanation
+	) {
+		Map<String, Object> model = new HashMap<>();
+		model.put(QueryResultView.FILENAME_HINT_KEY, "query-result");
+		model.put(QueryResultView.QUERY_RESULT_KEY, explanation);
```

Comment:
Maybe we want to introduce a new key. To make it less confusing. In theory we could at some point allow the query explanation and the query result to be shown at the same time.

Reply:
done